File: conformance/tests/blake3/blake3.runar.zig
const runar = @import("runar");

pub const Blake3Test = struct {
    pub const Contract = runar.SmartContract;

    expected: runar.ByteString,

    pub fn init(expected: runar.ByteString) Blake3Test {
        return .{ .expected = expected };
    }

    pub fn verifyCompress(self: *const Blake3Test, chainingValue: runar.ByteString, block: runar.ByteString) void {
        const result = runar.blake3Compress(chainingValue, block);
        runar.assert(runar.bytesEq(result, self.expected));
    }

    pub fn verifyHash(self: *const Blake3Test, message: runar.ByteString) void {
        const result = runar.blake3Hash(message);
        runar.assert(runar.bytesEq(result, self.expected));
    }
};

File: conformance/tests/blake3/blake3.runar.go
//go:build ignore

package contract

import "runar"

type Blake3Test struct {
	runar.SmartContract
	Expected runar.ByteString `runar:"readonly"`
}

func (c *Blake3Test) VerifyCompress(chainingValue runar.ByteString, block runar.ByteString) {
	result := runar.Blake3Compress(chainingValue, block)
	runar.Assert(result == c.Expected)
}

func (c *Blake3Test) VerifyHash(message runar.ByteString) {
	result := runar.Blake3Hash(message)
	runar.Assert(result == c.Expected)
}
